MoltenCore/Majordomo: Tank Teleport

Avyiel opened the pull request:
Closes #792.

This PR adds the `Teleport` skill that Majordomo casts on the Tank every 25~30 seconds. It has no message or warning sound, as it affects just the tank and requires almost no action.

elvador commented:
I'd recommend using CDBar instead of Bar for timers which aren't fully accurate.

See 
https://github.com/BigWigsMods/BigWigs/blob/classic/Core/BossPrototype.lua#L1993-L2029

Avyiel commented:
> I'd recommend using CDBar instead of Bar for timers which aren't fully accurate.
> 
> See
> https://github.com/BigWigsMods/BigWigs/blob/classic/Core/BossPrototype.lua#L1993-L2029

Ah, I'm slowly learning how to properly use the addon's API, thank you for the feedback. 

I've updated the "regular" teleport hook to use `CDBar` since the timer is variable and also added an exact `Bar` to `OnEngage`, since the first teleport is exactly after 20s.

elvador commented on MoltenCore/Majordomo.lua:
```diff
@@ -44,6 +44,10 @@ function mod:VerifyEnable(unit)
 end
 
+function mod:OnEngage()
+	self:Bar(20534, 20)
```
Please add the spell name as a comment for easy identification

Avyiel commented:
I just learned about option flags, should this `Teleport` be marked as `"TANK"`? 
```lua
{20534, "TANK"}, -- Teleport
```

elvador commented:
If the ability is only/primarily relevant to tanks, yes.

Non-tanks will always be able to enable it in the options.

Avyiel commented:
> If the ability is only/primarily relevant to tanks, yes.
> 
> Non-tanks will always be able to enable it in the options.

Well, the skill teleports the person with highest threat to the lava pit. So yeah, only relevant to tanks.
